volatility3/framework/symbols/windows/__init__.py

```diff
@@ -5,11 +5,12 @@
 
 from volatility3.framework.symbols import intermed
 from volatility3.framework.symbols.windows import extensions
-from volatility3.framework.symbols.windows.extensions import pe, pool, registry
 
 
 class WindowsKernelIntermedSymbols(intermed.IntermediateSymbolTable):
     def __init__(self, *args, **kwargs) -> None:
+        from volatility3.framework.symbols.windows.extensions import pe, pool, registry
```

Comment:
This is absolutely not the right way of solving this.  Can you let me know what the problem with the import is that required this move please?

Reply:
Ha, I had a feeling it was not. This is the original traceback:
```
ImportError: Failed to import test module: test_scheduled_tasks
Traceback (most recent call last):
  File "/Users/dmcdonald/.pyenv/versions/3.8.10/lib/python3.8/unittest/loader.py", line 154, in loadTestsFromName
    module = __import__(module_name)
  File "/Users/dmcdonald/volatility3/test/plugins/windows/test_scheduled_tasks.py", line 6, in <module>
    from volatility3.plugins.windows import scheduled_tasks
  File "/Users/dmcdonald/volatility3/volatility3/framework/plugins/windows/scheduled_tasks.py", line 18, in <module>
    from volatility3.framework.layers import registry
  File "/Users/dmcdonald/volatility3/volatility3/framework/layers/registry.py", line 17, in <module>
    from volatility3.plugins.windows import pslist
  File "/Users/dmcdonald/volatility3/volatility3/framework/plugins/windows/pslist.py", line 14, in <module>
    from volatility3.framework.symbols.windows.extensions import pe
  File "/Users/dmcdonald/volatility3/volatility3/framework/symbols/windows/__init__.py", line 8, in <module>
    from volatility3.framework.symbols.windows.extensions import pe, pool, registry
  File "/Users/dmcdonald/volatility3/volatility3/framework/symbols/windows/extensions/registry.py", line 11, in <module>
    from volatility3.framework.layers.registry import (
ImportError: cannot import name 'RegistryFormatException' from partially initialized module 'volatility3.framework.layers.registry' (most likely due to a circular import) (/Users/dmcdonald/volatility3/volatility3/framework/layers/registry.py)

```

Reply:
Ok, so the cycle is:

layers.registry -> windows.pslist -> symbols.windows...pe -> windows.extensions.{pe,pool,registry} -> layers.registry

Personally, I think the layers.registry should be able to work without pslist, so we need to investigate why that's there.  Core functionality *shouldn't* need a plugin to run...

Reply:
Apparently I added it 3 years ago because windows 10 now has a registry process...  5:S  So I'd almost rather manually implement finding the `Registry` process than pulling in the whole plugin and everything it brings with it, just to get the process list.

Fancy having a go at making that change and then pulling pslist out of the layers.registry code and documenting how it's duplicate code to avoid pulling in complex code?  Either that or we could carve the code out into its own file and import it into both to reduce code duplication?  I'm happy with either, it just walks the ActiveProcessLinks list from PsActiveProcessHead, and then uses that proc.  It can be as cut down as you like, that's literally the only thing it's doing...